File: exercises/beecrowd/1047/1047.ts
// https://judge.beecrowd.com/pt/problems/view/1047
import * as fs from "fs";

function calculatePeriod(beginning: number, end: number, limit: number): number {
	let duration = end - beginning
	if (duration < 0) {
		duration += limit
	}
	return duration
}

function main(): void {
	const input = fs.readFileSync("/dev/stdin", "utf8");
	const lines = input.split(/\r?\n/);

	const line = String(lines.shift()).split(' ');

	let a = Number(line[0])
	let b = Number(line[1])
	let c = Number(line[2])
	let d = Number(line[3])

	let hours = calculatePeriod(a, c, 24)
	let minutes = calculatePeriod(b, d, 60)

	if (a == c && b == d) {
		hours = 24
		minutes = 0
	} else if (b > d) {
		hours = (hours + 23) % 24
	}

	console.log(`O JOGO DUROU ${hours} HORA(S) E ${minutes} MINUTO(S)`)
}

main();

File: exercises/beecrowd/1047/1047.go
// https://judge.beecrowd.com.br/judge/pt/problems/view/1047
package main

import (
	"fmt"
)

func calculatePeriod(beginning, end, limit int) int {
	duration := end - beginning
	if duration < 0 {
		duration += limit
	}
	return duration
}

func main() {
	var a, b, c, d int

	fmt.Scanf("%d %d %d %d", &a, &b, &c, &d)

	hours := calculatePeriod(a, c, 24)
	minutes := calculatePeriod(b, d, 60)

	if a == c && b == d {
		hours = 24
		minutes = 0
	} else if b > d {
		hours = (hours + 23) % 24
	}

	fmt.Printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", hours, minutes)
}
